src/sensors/LaserRangeFinder.cpp

```diff
@@ -298,116 +300,34 @@ void LaserRangeFinder::setNumBeams(int num_beams) {
-    // smallest coordinate by the biggest).
-    for(int j = 0; j < 8; ++j)
-    {
-        slope_to_index_[j].resize(N);
-        for(int k = 0; k < N; ++k)
-        {
-            // Calculate the slope corresponding to this table entry
-            double slope = (double)k / slope_factor_;
-
-            // Calculate a virtual (x, y) cartesian point which corresponds to this table entry
-            // based on the slope. Determine the part of the unit circle in which (x, y) lies
-            // base on j
-            double x, y;
-            if (j & 1)
-            {
-                // x_abs > y_abs
-                x = 1;
-                y = slope + 1e-16; // Necessary because of 0-check in getAngle
-            }
-            else
-            {
-                // x_abs < y_abs
-                x = slope + 1e-16; // Necessary because of 0-check in getAngle
-                y = 1;
-            }
-
-            if (!(j & 2))
-            {
-                // x < 0
-                x = -x;
-            }
-
-            if (!(j & 4))
-            {
-                // y < 0
-                y = -y;
-            }
-
-            // Calculate the angle of the virtual point (x, y)
-            double a = getAngle(x, y);
-            double a_temp = a - a_min_;
-            if (a_temp < 0)
-                a_temp += 2 * M_PI;
-
-            // Calculate the ray index based on the angle
-            slope_to_index_[j][k] = a_temp / (a_max_ - a_min_) * num_beams_ + 1;
-        }
+    for(uint i = 0; i < num_beams_; ++i) {
+        ray_dirs_[i] = polarTo2D(a, 1);
+        angles_[i] = a;
+        a += angle_incr_;
     }
 
-    i_half_circle_ = M_PI / getAngleIncrement();
+    i_half_circle_ = M_PI / angle_incr_;
 }
 
 double LaserRangeFinder::getAngleIncrement() const {
-    return (a_max_ - a_min_) / std::max(num_beams_ - 1, 1);
+    return angle_incr_;
 }
 
-int LaserRangeFinder::getAngleUpperIndex(double angle) const {
-    int i = (angle - a_min_) / (a_max_ - a_min_) * num_beams_ + 1;
-    return std::min(num_beams_, std::max(0, i));
+uint LaserRangeFinder::getAngleUpperIndex(double angle) const {
+    int i = (angle - a_min_) / angle_incr_ + 1;
+    return std::min<uint>(num_beams_, std::max<int>(0, i));
 }
 
-int LaserRangeFinder::getAngleUpperIndex(double x, double y) const
-{
+uint LaserRangeFinder::getAngleUpperIndex(double x, double y) const {
     // Calculate the ray index corresponding to the cartesian point (x, y)
-    // We use the look-up table to avoid calculating the atan of (x / y)
-
-    double x_abs = std::abs(x);
-    double y_abs = std::abs(y);
-
-    // Calculcate slope
-    double slope;
-    if (x_abs < y_abs)
-        slope = x_abs / y_abs;
-    else
-        slope = y_abs / x_abs;
-
-    int k = slope_factor_ * slope;
-
-    // Calculate region number (0 - 7)
-    int j = (x_abs < y_abs ? 0 : 1) + (x < 0 ? 0 : 2) + (y < 0 ? 0 : 4);
-
-    return slope_to_index_[j][k];
+    return getAngleUpperIndex(atan2(y, x));
```

Comment:
Or is this the most important change? 

From the comments it seems, however, that the most important change here is the use of the atan2 function instead of a look-up table.

Reply:
Yes, indeed switching from this look-up table to atan2. But also the change in the angle function. As the look-up table behaviour matched the behaviour of the angle function.